src/badge/badge.style.tsx

```diff
@@ -0,0 +1,95 @@
+import styled, { css } from "styled-components";
+import { Border, Colour, Font, Radius } from "../theme";
+import { BadgeProps } from "./types";
+
+// =============================================================================
+// STYLE INTERFACE, transient props are denoted with $
+// See more https://styled-components.com/docs/api#transient-props
+// =============================================================================
+interface StyledBadgeProps {
+    $variant: BadgeProps["variant"];
+    $color: BadgeProps["color"];
+}
+interface BadgeContainerProps {
+    $isOverlay?: boolean;
+}
+interface BadgeWrapperProps extends BadgeContainerProps {
+    $offset?: string;
+}
+
+// =============================================================================
+// STYLING
+// =============================================================================
+export const BadgeOverlay = styled.div<BadgeContainerProps>`
+    ${(props) =>
+        props.$isOverlay &&
+        css`
+            position: relative;
+            width: fit-content;
+            height: fit-content;
+        `}
+`;
+
+export const BadgeWrapper = styled.div<BadgeWrapperProps>`
+    ${(props) =>
+        props.$isOverlay &&
+        css`
+            position: absolute;
+            top: 0;
+            right: 0;
+            transform: translate(${props.$offset ?? "50%, -25%"});
```

Comment:
suggestion: support `badgeOffset` prop as `[number | string, number | string]`

in the transform, we can apply this offset relative to the default position
```
transform: translate(50%, -25%)
    ${props.$offset
        ? `translate(${props.$offset[0]},${props.$offset[1]})`
        : ""};
```

inspired by the behaviour of [MUI](https://mui.com/joy-ui/react-badge/#inset) and [Ant Design](https://ant.design/components/badge#badge-demo-offset)